src/Bridge/Doctrine/Orm/Extension/PaginationExtension.php

```diff
@@ -33,28 +32,14 @@
     private $managerRegistry;
     private $requestStack;
     private $resourceMetadataFactory;
-    private $enabled;
-    private $clientEnabled;
-    private $clientItemsPerPage;
-    private $itemsPerPage;
-    private $pageParameterName;
-    private $enabledParameterName;
-    private $itemsPerPageParameterName;
-    private $maximumItemPerPage;
-
-    public function __construct(ManagerRegistry $managerRegistry, RequestStack $requestStack, ResourceMetadataFactoryInterface $resourceMetadataFactory, bool $enabled = true, bool $clientEnabled = false, bool $clientItemsPerPage = false, int $itemsPerPage = 30, string $pageParameterName = 'page', string $enabledParameterName = 'pagination', string $itemsPerPageParameterName = 'itemsPerPage', int $maximumItemPerPage = null)
+    private $paginationHelper;
+
+    public function __construct(ManagerRegistry $managerRegistry, RequestStack $requestStack, ResourceMetadataFactoryInterface $resourceMetadataFactory, PaginationHelper $paginationHelper)
```

Comment:
Yes, this would not be BC. But on the other side it would not make sense to keep these variables in the constructor.